### ISSUE TITLE
[Bug][Jupyter Widget] - Model class 'KeplerGlModal' from module 'keplergl-jupyter' is loaded but can not be instantiated

### ISSUE THREAD

**Issue by ManuelBanza** [keplergl/kepler.gl#1993]
Hi,

I do not know what happen but know I can not create or load maps. This message appears:

![imagem](https://user-images.githubusercontent.com/53525022/192508067-8f8d0f6e-9e10-46a0-93fd-a149c46d0633.png)

![imagem](https://user-images.githubusercontent.com/53525022/192508130-d37e1fee-7645-49fc-88fe-d0d2d3a267ac.png)

I uninstall and re install it all over again, and also on a new python environment but the problem remains.

Can anyone help?

Cheers

**Comment by jenks-guo-filecoin** [comment]
Same problem here. 

![image](https://user-images.githubusercontent.com/100632895/195980732-acbbc702-ba76-4f67-a56e-291edf75dcf8.png)

![image](https://user-images.githubusercontent.com/100632895/195980802-b92bf5c0-e240-4036-b704-5c239a67bb2f.png)

**Comment by ManuelBanza** [comment]
Any news about this? I have tried everything but the problem remains...

**Comment by rloredo** [comment]
Same problem here, only that I'm inside a docker container

**Comment by fboerman** [comment]
I too have this issue and it completely breaks the jupyter widget :(

**Comment by Nicole-Hong** [comment]
I am also struggling with the same issue. I think the problem is related to Node.js and Jupyterlab extension, so I followed the instruction for Windows in this link: https://nodejs.org/en/download/package-manager/#windows-1  I got stuck with the installation guide under "Chocolately" and "Scoop" and still could not resolve this issue. I am wondering if JupyterLab is not compatible with keplergl at all...how can I resolve this issue?

**Comment by lennonzheng** [comment]
I would move to other mapping platform.. (Dash / Folium) as Kepler's last release was in 2021 https://github.com/keplergl/kepler.gl/releases/tag/v2.5.5

**Comment by rloredo** [comment]
@lennonzheng that is true, but there is still activity in the repo. The latest commit was 12hs ago.

**Comment by AliBashiri** [comment]
any updates on this?

**Comment by fboerman** [comment]
bump, I really would like to use this but cant right now

**Comment by gc1201** [comment]
same issue here, any updates?

**Comment by juliocraf** [comment]
Hey guys.

I followed the following steps and it worked.

I use Windows 10 and in my case the problem was the node.js that was not installed.

the steps I followed were:

conda install -c conda-forge keplergl


jupyter nbextension install --py --sys-prefix keplergl # can be skipped for notebook 5.3 and above


jupyter nbextension enable --py --sys-prefix keplergl # can be skipped for notebook 5.3 and above

I installed node.js and it worked normally


![print](https://user-images.githubusercontent.com/50004681/222748108-2d7c13c1-5b6b-42eb-a5d1-6a5aefcc8a5b.png)

**Comment by curtis18** [comment]
@juliocraf I tried, but not working. May I know your python and node.js version? Thank you.

**Comment by juliocraf** [comment]
"Good morning, I used this step-by-step guide here and it worked. I was getting the same error messages. The versions are: node 18.14.2,  python: 3.9.16 abd keplergl: 0.3.2"

**Comment by curtis18** [comment]
@juliocraf It works now, thanks a lot. I think it is related to the version of ipywidgets, and widgetsnbextension. There may be problem if ipywidgets >= 8.1 and widgetsnbextension >= 4.0. Using ipywidgets 7.8.0 and widgetsnbextension 3.6.5 works normal now.

**Comment by webocean8** [comment]
It works now, thanks a lot.

I am using the node.js version v18.17.1
IPython : 8.15.0
ipykernel : 6.25.2
ipywidgets : 8.1.0
jupyter_client : 7.4.9
jupyter_core : 5.3.1
jupyter_server : 2.7.3
jupyterlab : 3.6.5

Using ipywidgets 7.8.0 and widgetsnbextension 3.6.5 works normal now.
thanks

**Comment by Cheese-Bar** [comment]
Thanks a lot. I spend a whole night to find the solution, and now it works on jupyter notebook.
I try a lot, but it still not work in jupyter lab, although with the same error message.

Anyway, the fllowing setting makes it work on **jupyter notebook.** It may be helpful.

node.js v18.14.2
ipython 8.15.0
ipykernel 6.25.2
ipywidgets 7.8.0
widgetsnbextension 3.6.5
jupyter_client 7.4.9
jupyter_core 5.3.1
jupyter_server 2.7.3
jupyterlab 3.6.5
nbclient 0.8.0
nbconvert 7.8.0
nbformat 5.9.2
notebook 6.5.5
traitlets 5.9.0

![image](https://github.com/user-attachments/assets/583b2e17-3a92-453a-ac5a-ae33dd183d6c)
